tool/tsh/putty_config_windows.go

```diff
@@ -0,0 +1,313 @@
+			if !slices.Contains(hostList, wildcardDomain) {
+				log.Debugf("Adding wildcard %s to hostList", wildcardDomain)
+				hostList = append(hostList, wildcardDomain)
+			} else {
+				log.Debugf("Not adding %s because it's already covered by %s", host, wildcardDomain)
+				continue
+			}
+		} else {
+			if !slices.Contains(hostList, host) {
+				log.Debugf("Adding %s to hostList", host)
+				hostList = append(hostList, host)
+			}
+		}
+	}
+	sort.Strings(hostList)
+
+	// write strings to subkey
+	if ok, err := registryWriteMultiString(registryKey, "MatchHosts", hostList); !ok {
+		return false, trace.Wrap(err)
+	}
+	if ok, err := registryWriteString(registryKey, "PublicKey", publicKey); !ok {
+		return false, trace.Wrap(err)
+	}
+
+	return true, nil
+}
+
+// formatLocalCommandString replaces placeholders in a constant with actual values
+func formatLocalCommandString(tshPath string, cluster string) string {
+	// replace the placeholder "%cluster" with the actual cluster name as passed to the function
+	clusterString := strings.ReplaceAll(puttyProxyTelnetCommand, "%cluster", cluster)
+	// PuTTY needs its paths to be double-escaped i.e. C:\\Users\\User\\tsh.exe
+	escapedTshPath := strings.ReplaceAll(tshPath, `\`, `\\`)
+	return strings.ReplaceAll(clusterString, "%tsh", escapedTshPath)
+}
+
+// onPuttyConfig handles the `tsh config putty` subcommand
+func onPuttyConfig(cf *CLIConf) error {
+	if runtime.GOOS != constants.WindowsOS {
+		return trace.NotImplemented("PuTTY config is only implemented on Windows")
+	}
+
+	tc, err := makeClient(cf, true)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	// connect to proxy to fetch cluster info
+	proxyClient, err := tc.ConnectToProxy(cf.Context)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	defer proxyClient.Close()
+
+	// parse out proxy details
+	proxyHost, _, err := net.SplitHostPort(tc.Config.SSHProxyAddr)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	// get the public key of the Teleport host CA so we can add it to PuTTY's list of trusted keys
+	var hostCAKnownHostsLine string
+	err = tc.WithRootClusterClient(cf.Context, func(clt auth.ClientI) error {
+		hostCAKnownHostsLine, err = client.ExportAuthorities(cf.Context, clt, client.ExportAuthoritiesRequest{AuthType: "host"})
+		if err != nil {
+			return trace.Wrap(err)
+		}
+		return nil
+	})
+	_, _, hostCAPublicKeyBytes, _, _, err := ssh.ParseKnownHosts([]byte(hostCAKnownHostsLine))
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	hostCAPublicKey := strings.TrimSuffix(string(ssh.MarshalAuthorizedKey(hostCAPublicKeyBytes)), "\n")
+
+	// get root cluster name
+	rootClusterName, rootErr := proxyClient.RootClusterName(cf.Context)
+	// TODO(gus): figure out what to do (if anything) about leaf clusters
+	//leafClusters, leafErr := proxyClient.GetLeafClusters(cf.Context)
+	_, leafErr := proxyClient.GetLeafClusters(cf.Context)
+	if err := trace.NewAggregate(rootErr, leafErr); err != nil {
+		return trace.Wrap(err)
+	}
```

Comment:
Given what we've seen with Machine ID feedback, leaf clusters don't come up too often but definitely will come up eventually.

I assume in PuTTY's case it'd mean inserting one entry per (leaf) cluster into the registry, and into PuTTY's list of destinations? If we just copied what `tsh config` does (generating configs for root + all leafs) that could be a bit spammy for users with lots of leaf clusters. Maybe we can have a `--leaf <name>` flag or something to insert them one at a time as desired, and just override the cluster name we insert?